app/scripts/controllers/reviewRegistration.js

```diff
@@ -68,6 +68,182 @@ angular.module('confRegistrationWebApp')
+        modalMessage.error(data.error ? data.error.message : 'An error occurred while submitting your registration.');
+      });
+    };
+
+    ////// START EVENT-433 //////
+
+    if (!$rootScope.includeSpouseRegistrationClicked) {
+      // Get spouse registration
+      // End point: getSpouseRegistration = /conferences/conferenceId/registrations/spouse
+      ///// TEST DATA end point url: /conferences/conferenceId/registrations/current /////
+      $http.get('conferences/' + conference.id + '/registrations/current')
+        .then(function(response) {
+          if (response.status === 200) {
+            $scope.spouseRegistration = response.data;
+
+            ///// Test Data /////
+            var id = uuid();
+            $scope.spouseRegistration.registrants[0].id = id;
+            $scope.spouseRegistration.registrants[0].registrationId = uuid();
+            $scope.spouseRegistration.registrants[0].userId = uuid();
+            $scope.spouseRegistration.registrants[0].email = 'janeworner3@hotmail.com';
+            $scope.spouseRegistration.registrants[0].firstName = 'Jane';
+            $scope.spouseRegistration.registrants[0].lastName = 'Worner3';
+            $scope.spouseRegistration.registrants[0].answers[1] = {id: uuid(), registrantId: id, blockId: 'b56c9fb7-9643-47d9-9408-ead1bcd6ab31', value: 'janeworner3@hotmail.com'};
+            $scope.spouseRegistration.registrants[0].answers[0] = {id: uuid(), registrantId: id, blockId: '407248f1-6244-49e7-964f-5508634d5ce5', value: {firstName: 'Jane', lastName: 'Worner3'}};
+            ///// Test Data /////
+
+            $scope.spouseExists = true;
+            $rootScope.includeSpouseButtonVisible = true;
+          }
+          else {
+            $scope.spouseRegistration = {};
+            $scope.spouseExists = false;
+            $rootScope.includeSpouseButtonVisible = false;
+            console.log('No current and/or spouse registration found for user with ID.');
+          }
+        }).catch(function(response) {
+          $scope.spouseRegistration = {};
+          $rootScope.includeSpouseButtonVisible = false;
+          console.log('No current and/or spouse registration found for user with ID. Status = ' + response.status);
+      });
+    }
+
+    //Parameters needed to retrieve all current registrants for conference.id
+    var queryParameters = {
+      block: [],
+      //page: 1,
+      //limit: 20,
+      orderBy: 'last_name',
+      order: 'ASC',
+      filter: '',
+      filterPayment: '',
+      filterRegType: '',
+      includeCheckedin: 'yes',
+      includeWithdrawn: 'yes',
+      includeIncomplete: 'yes'
+    };
+
+    // Spouse Registration section visibility
+    $scope.spouseSectionVisible = {
+      spouse: false,
+      current: false,
+      both: false,
+      neither: false
+    };
+
+    //Get all current registrants based on conference.id and queryParameters
+    $http.get('conferences/' + conference.id + '/registrations', {params: queryParameters})
```

Comment:
This introduces a race condition. If this request completes before the current registrations request, then `$scope.spouseRegistration` will be undefined, and trying to access the `registrants` property will throw an exception.